app/controllers/concerns/with_certificate_render.rb

```diff
@@ -0,0 +1,25 @@
+require 'wicked_pdf'
+require 'rqrcode'
+
+module WithCertificateRender
+
+  extend ActiveSupport::Concern
+
+  included do
+    helper_method :qr_for
+  end
+
+  def qr_for(certificate)
+    qr = RQRCode::QRCode.new(verify_certificate_url certificate.code).as_svg(color: '0B465D')
+    "data:image/svg+xml,#{URI.encode(qr)}"
+  end
+
+  def pdf_for(certificate)
```

Comment:
Both PDF and QR should be generated only once and then saved somewhere. 
This seems like a good use case for an _S3 like storage_

Reply:
Of course this could be left as an enhancement for later!